packages/monaco/src/browser/monaco-frontend-module.ts

```diff
@@ -190,11 +178,34 @@ export function createMonacoConfigurationService(container: interfaces.Container
         }
         return sections;
     };
+
+    const toTarget = (scope: PreferenceScope): monaco.services.ConfigurationTarget => {
+        switch (scope) {
+            case PreferenceScope.Default: return monaco.services.ConfigurationTarget.DEFAULT;
+            case PreferenceScope.User: return monaco.services.ConfigurationTarget.USER;
+            case PreferenceScope.Workspace: return monaco.services.ConfigurationTarget.WORKSPACE;
+            case PreferenceScope.Folder: return monaco.services.ConfigurationTarget.WORKSPACE_FOLDER;
+        }
+    };
+
     preferences.onPreferencesChanged((changes?: PreferenceChanges) => {
         const affectedSections = parseSections(changes);
+
+        let target: monaco.services.ConfigurationTarget | undefined;
+        if (changes && affectedSections && affectedSections.length > 0) {
+            const change = changes[affectedSections[0]];
+            target = toTarget(change.scope);
+        }
+
         if (affectedSections) {
-            const event = new monaco.services.ConfigurationChangeEvent();
-            event.change(affectedSections);
+            const keys = affectedSections || [];
+            const previous = { data: service._configuration.toData() };
```

Comment:
we cannot use `toData`, the goal here is to override methods of configuration and delegate to our preference service. `toData` will simply return default Monaco values